va/validation-authority.go

```diff
@@ -275,6 +279,14 @@ func (va *ValidationAuthorityImpl) fetchHTTP(ctx context.Context, identifier cor
 			Detail: fmt.Sprintf("Error reading HTTP response body: %v", err),
 		}
 	}
+	// io.LimitedReader will silently truncate a Reader so if the
+	// resulting payload is the same size as maxResponseSize fail
+	if len(body) >= maxResponseSize {
+		return nil, validationRecords, &probs.ProblemDetails{
+			Type:   probs.UnauthorizedProblem,
+			Detail: fmt.Sprintf("Invalid response from %s: response too large, \"%s\"", url.String(), body),
```

Comment:
I'm not sure what you mean by "doesn't match", the only thing that can cause this to be returned is the response being >= 128 bytes, `fetchHTTP` does no actual checking of the contents of the payload.